mne/viz/ui_events.py

```diff
@@ -263,9 +264,13 @@ def delete_event_channel(event=None, *, weakfig=weakfig):
         # Hook up the above callback function to the close event of the figure
         # window. How this is done exactly depends on the various figure types
         # MNE-Python has.
-        _validate_type(fig, (matplotlib.figure.Figure, Brain, EvokedField), "fig")
+        _validate_type(
+            fig, (matplotlib.figure.Figure, Brain, EvokedField, BrowserBase), "fig"
+        )
         if isinstance(fig, matplotlib.figure.Figure):
             fig.canvas.mpl_connect("close_event", delete_event_channel)
+        elif isinstance(fig, BrowserBase):
+            fig.mne.viewbox.destroyed.connect(delete_event_channel)
```

Comment:
this line does not work for the matplotlib-based browser. The code will actually work, because the matplotlib-based browser case is handled in the if-statement above, but let's make that more explicit.

```suggestion
            # The matplotlib-based browser case is a matplotlib.figure.Figure so is already
            # handled. Therefore we can assume this is the QT-based browser.
            from mne_qt_browser._pg_figure import MNEQtBrowser
            assert isinstance(fig, MNEQtBrowser)
            fig.mne.viewbox.destroyed.connect(delete_event_channel)           
```